grug/ffi/src/exports.rs

```diff
@@ -43,12 +43,14 @@ where
     let res = (|| {
         let ctx: Context = unwrap_into_generic_result!(ctx_bytes.deserialize_borsh());
         let ctx = make_mutable_ctx!(ctx, &mut ExternalStorage, &ExternalApi, &ExternalQuerier);
-        let msg = unwrap_into_generic_result!(msg_bytes.deserialize_json());
+
+        let msg = unwrap_into_generic_result!(msg_bytes.deserialize_borsh::<Json>());
+        let msg = unwrap_into_generic_result!(msg.deserialize_json());
```

Comment:
Now we need two deserialize steps:

bytes --[borsh]--> `serde_json::Value` --[json]--> `M`

But this is still faster than before.